modin/pandas/test/dataframe/test_map_metadata.py

```diff
@@ -638,6 +638,20 @@ def test_convert_dtypes_multiple_row_partitions():
     assert modin_result.dtypes.equals(pandas_result.dtypes)
 
 
+@pytest.mark.skipif(
+    get_current_execution() == "BaseOnPython",
+    reason="BaseOnPython cannot not have multiple partitions.",
+)
+def test_convert_dtypes_5653():
+    modin_part1 = pd.DataFrame({"col1": ["a", "b", "c", "d"]})
+    modin_part2 = pd.DataFrame({"col1": [None, None, None, None]})
+    modin_df = pd.concat([modin_part1, modin_part2])
+    assert modin_df._query_compiler._modin_frame._partitions.shape == (2, 1)
```

Comment:
neither HDK nor BaseOnPython has partitions, so let's make this check more accurate
```suggestion
    if StorageFormat.get() == "Pandas"
        assert modin_df._query_compiler._modin_frame._partitions.shape == (2, 1)
```